integrations/woocommerce/rcp-restrict-woocommerce-shop-archive.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Plugin Name: Restrict Content Pro - Restrict WooCommerce Shop Archive
+ * Description: Restricts access to the WooCommerce shop archive, using the settings specified in the "Restrict this Content" metabox.
+ * Version: 1.0
+ * Author: Restrict Content Pro team
+ * License: GPL2
+ */
+
+/**
+ * Restricts access to the WooCommerce shop archive according to the settings specified in the metabox.
+ * If the current user doesn't have access, they're redirected to the specified redirect URL in Restrict > Settings > Misc.
+ *
+ * @since 1.0
+ * @return void
+ */
+function ag_rcp_redirect_woocommerce_shop() {
+	if ( ! function_exists( 'is_shop' ) || ! is_shop() ) {
+		return;
+	}
```

Comment:
Need to check for `RCP_Member` here too, in case someone deactivates RCP.

Reply:
Updated!